lynxkite-app/web/src/workspace/Workspace.tsx

```diff
@@ -247,7 +252,7 @@ function LynxKiteFlow() {
         event.preventDefault();
         setNodeSearchSettings({
           pos: getBestPosition(),
-          boxes: catalog.data![state.workspace.env!],
+          categoryHierarchy: buildCategoryHierarchy(catalog.data![state.workspace.env!]),
```

Comment:
You're still re-building the hierarchy each time the search is opened. Build it when we receive it from the backend.